bitcoin/src/p2p/mod.rs

```diff
@@ -262,18 +264,20 @@ macro_rules! generate_network_magic_conversion {
             fn try_from(magic: Magic) -> Result<Self, Self::Error> {
                 match magic {
                     $(
-                        Magic::$magic => Ok(Network::$network),
+                        Magic::$magic => Ok(Network::$network$((TestnetVersion::$testnet_version))?),
                     )*
                     _ => Err(UnknownMagicError(magic)),
                 }
             }
         }
     };
 }
-
+// Generate conversion functions for all known networks.
+// `Network -> Magic` and `Magic -> Network`
 generate_network_magic_conversion! {
     Network::Bitcoin => Magic::BITCOIN,
-    Network::Testnet => Magic::TESTNET,
+    Network::Testnet(TestnetVersion::V3) => Magic::TESTNET,
```

Comment:
Should we deprecate `Magic::TESTNET` also and add `Magic::TESTNET3` ?

Reply:
Oh, yeah, good idea.